worlds/messenger/__init__.py

```diff
@@ -116,44 +210,132 @@ def create_items(self) -> None:
+                spoiler_handle.write(f"{portal}\n")
+
+        spoiler = self.multiworld.spoiler
+
+        if self.options.shuffle_portals:
+            # sort the portals as they appear left to right in-game
+            portal_info = sorted(
+                self.spoiler_portal_mapping.items(),
+                key=lambda portal:
+                ["Autumn Hills", "Riviere Turquoise",
+                 "Howling Grotto", "Sunken Shrine",
+                 "Searing Crags", "Glacial Peak"].index(portal[0]))
+            for portal, output in portal_info:
+                spoiler.set_entrance(f"{portal} Portal", output, "I can write anything I want here lmao", self.player)
+
+        # if self.options.shuffle_transitions:
+        #     for transition in self.transitions:
+        #         if (transition.er_type == Entrance.EntranceType.TWO_WAY and
+        #                 (transition.connected_region.name, "both", self.player) in spoiler.entrances):
+        #             continue
+        #         spoiler.set_entrance(
+        #             transition.name if "->" not in transition.name else transition.parent_region.name,
+        #             transition.connected_region.name,
+        #             "both" if transition.er_type == Entrance.EntranceType.TWO_WAY
+        #                       and self.options.shuffle_transitions == ShuffleTransitions.option_coupled
+        #             else "",
+        #             self.player)
+
+    # def extend_hint_information(self, hint_data: Dict[int, Dict[int, str]]) -> None:
+    #     if not self.options.shuffle_transitions:
+    #         return
+    # 
+    #     hint_data.update({self.player: {}})
+    # 
+    #     all_state = self.multiworld.get_all_state(True)
+    #     # sometimes some of my regions aren't in path for some reason?
+    #     all_state.update_reachable_regions(self.player)
+    #     paths = all_state.path
+    #     start = self.multiworld.get_region("Tower HQ", self.player)
+    #     start_connections = [entrance.name for entrance in start.exits if entrance not in {"Home", "Shrink Down"}]
+    #     transition_names = [transition.name for transition in self.transitions] + start_connections
+    #     for loc in self.multiworld.get_locations(self.player):
+    #         if (loc.parent_region.name in {"Tower HQ", "The Shop", "Music Box", "The Craftsman's Corner"}
+    #                 or loc.address is None):
+    #             continue
+    #         path_to_loc = []
+    #         name, connection = paths[loc.parent_region]
+    #         while connection != ("Menu", None):
+    #             name, connection = connection
+    #             if name in transition_names:
+    #                 path_to_loc.append(name)
+    # 
+    #         text = ""
+    #         for transition in reversed(path_to_loc):
+    #             text += f"{transition} => "
+    #         text = text.rstrip("=> ")
+    #         hint_data[self.player][loc.address] = text
```

Comment:
the main reason i didn't is because the features were kind of developed in tandem so i don't have a clean git history to just move them with. moving them out of here completely wouldn't be too difficult to do, though a bit of a pain to attempt to keep it clean. would prefer to leave it as is, but will remove it if a core maintainer says to.